tests/pages/courses/id.test.js

```diff
@@ -0,0 +1,169 @@
+import { auth } from '../../../firebase/initFirebase' // Import Firebase authentication
+import { SessionProvider } from 'next-auth/react' // Import the SessionProvider from next-auth for handling authentication state
+
+// Mock external dependencies to isolate and control their behavior during tests
+jest.mock('react-i18next', () => ({
+  useTranslation: () => ({
+    t: (key) => key, // Mock translation function that returns the translation key
+    i18n: { resolvedLanguage: 'en' }, // Mock i18n object with the resolved language
+  }),
+}))
+
+jest.mock('next/router', () => ({
+  useRouter: () => ({ query: {} }), // Mock the Next.js router hook with an empty query object
+}))
+
+// Mocking external libraries for backend operations and components
+jest.mock('../../../lib/course')
+jest.mock('../../../lib/cohorts')
+jest.mock('../../../lib/lessons')
+jest.mock('../../../lib/user')
+
+// Mocking the Next.js <Head> component
+jest.mock('next/head', () => ({
+  __esModule: true,
+  default: ({ children }) => <>{children}</>, // Mock the <Head> component to avoid issues with it in the tests
+}))
+
+describe('Course page when not yet started', () => {
+  // Mock data representing a course and a cohort
+  const mockCourse = {
+    id: 'Rust_State_Machine',
+    active: true,
+    title: 'Test Course',
+    description: 'A test course description',
+    sections: {
+      Section_1: [
+        { title: 'Lesson 1', file: 'lesson-1' },
+        { title: 'Lesson 2', file: 'lesson-2' },
+      ],
+    },
+  }
+
+  const mockCohort = {
+    id: 'RU5mLpQrZZWlmftNSB2w',
+    startDate: new Date(Date.now() + 86400000 * 7).getTime(), // Mock cohort starting date 7 days from now
+    endDate: new Date(Date.now() + 86400000 * 7 + 3600000).getTime(), // Mock cohort end date 1 hour after start
+    kickoffStartTime: new Date(Date.now() + 86400000 * 7).getTime(), // Mock cohort kickoff start time
+    kickoffEndTime: new Date(Date.now() + 86400000 * 7 + 3600000).getTime(), // Mock cohort kickoff end time
+  }
+
+  const mockUser = {
+    uid: 'user-123',
+    cohort_ids: [], // User initially not part of any cohort
+    cohorts: [],
+  }
+
+  // Before each test, reset all mock functions and set the default mock values
+  beforeEach(() => {
+    jest.clearAllMocks()
+    getCourse.mockResolvedValue(mockCourse) // Mock getCourse function to return mock course data
+    getFieldContent.mockImplementation((obj, field) => obj[field]) // Mock field content retrieval
+    getAllCohorts.mockResolvedValue([mockCohort]) // Mock getAllCohorts function to return a list of cohorts
+    getCurrentCohort.mockReturnValue(mockCohort) // Mock getCurrentCohort function to return the current cohort
+    getLessonsSubmissions.mockResolvedValue([]) // Mock lesson submissions
+    getUserFromFirestore.mockResolvedValue(mockUser) // Mock user data retrieval from Firestore
```

Comment:
same here at all these function calling. You can use one single comment to explain what `function_name.mockResolvedValue` and/or `function_name.mockReturnValue` do and let code cleaner.